### DISCUSSION TITLE
Query Parameters don't pass lists literally

### DISCUSSION THREAD

**Discussion by ghost** [encode/httpx#1596] [Potential Issue] [answered]
Apologies in advance if this is my error.

An API I'm dealing with expects a list as a parameter argument, for example: ``https://www.example.com/endpoint?locale=en_US&range=[0,999]``

Calling
```Python
httpx.get("https://www.example.com/endpoint", params={'locale':'en_US', 'range': [0,999]})
```

Unfortunately resolves to ``<Request('GET', 'https://www.example.com/endpoint?locale=en_US&range=0&range=999')>``

Am I doing something incorrectly or is this an issue?

Thank you.

**Comment by johtso** [comment, marked as answer]
HTTPX is just trying to be helpful and represent your list in the proper query parameter fashion, which is to declare the same key multiple times with each value.

If you want to avoid this behaviour, just turn your list into a string in the format you want it to be transmitted in :)

**Reply by ghost** [reply]
Hey johtso, 

Thanks for the quick reply - appreciate that does solve it (and I should've included it in my original post), was just trying to establish whether it was expected behavior?

Thanks,

**Reply by lovelydinosaur** [reply]
Yup, it's expected. I *assume* you'd get the same behaviour with `requests` as well, but haven't double checked.
